ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -512,10 +515,20 @@ public void delete(List<MessageInfo> infosToDelete) throws StoreException {
       List<IndexValue> indexValuesToDelete = new ArrayList<>();
       List<Short> lifeVersions = new ArrayList<>();
       Offset indexEndOffsetBeforeCheck = index.getCurrentEndOffset();
+      maybeCallInDeleteBetweenGetEndOffsetAndFindKey();
       for (MessageInfo info : infosToDelete) {
         IndexValue value = index.findKey(info.getStoreKey());
-        if (value == null) {
-          throw new StoreException("Cannot delete id " + info.getStoreKey() + " since it is not present in the index.",
+        // TODO: Passing a FileSpan the findKey and change the findKey implementation to fully restrict the index
+        // searching space to be limited within the given FileSpan.
+        if (value == null || (value.getOffset().compareTo(indexEndOffsetBeforeCheck) >= 0 && value.isPut())) {
```

Comment:
There could be one other edge case here where both a put and a ttl update are present after `indexEndOffsetBeforeCheck`. In this case a put entry will be returned with the "Ttl_Update_Index" flag set, which will make `value.isPut()` return false (See https://github.com/linkedin/ambry/blob/master/ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java#L629)

I suppose this may be even lower probability so we may not need to worry about this in this temporary fix. However, at some point its probably worth looking back on the IndexValue.isPut method (i.e. does a put with ttlupdate flag set by the index logic count as a put record?).

Reply:
yeah, this is caused by a slight different rules when reading the indexvalue than write the indexvalue. When writing index value, PUT and TTL_UPDATE are different records. While reading indexvalue, TTL_UPDATE will be merged with other records. It becomes just a flag. 

I suppose we don't have to worry about it at this PR, since next PR, I will use a FileSpan to restrict the search space in findKey.